### PR TITLE
Fix blocks2image height crop issue

### LINKED ISSUE #5165: blocks2image : When SVG is output, the definition block is not visible.
### Description

When I output SVG with blocks2image, the definition block is not visible.

image
The top of the block is not visible except for the definition block.

image

However, the definition block is not seen through.

js:blocks2image>userscript.js 180line
```
      `translate(0,${dataShapes === "hat" ? "18" : dataShapes=== "c-block c-1 hat" ? 21 : 0}) ${isExportPNG ? "scale(2)" : ""}`
```
It would seem to be a good idea to modify it in this way.

### Steps to reproduce

1.
2.
3.

### Demonstration(s) and additional context

_No response_

### Environment

Browser: 
Scratch Addons version: 
Operating system:

### PR DESCRIPTION
**Pull request by WorldLanguages**
I'm making this PR on behalf of @dos-pasos

Resolves #5165

### Changes

See https://github.com/ScratchAddons/ScratchAddons/discussions/5601

### CODE REVIEW

**Comment by dos-pasos** [comment]
> Probably fixes #5165.

Confirmo

**Comment by Samq64** [comment]
When exported as an SVG the block stack is positioned a little too low causing the bottom to be cut off with some image viewers. (Don't worry about the text, I just don't have the font installed)

image

**Review by Samq64 — CHANGES_REQUESTED** [review]
See my above comment.

**Comment by dos-pasos** [comment]
> When exported as an SVG the blocks are positioned too low causing the bottom to be cut off in some image viewers. (don't worry about the text, I just don't have the right font installed)
> 
> image
El problema es por la scala, agregué una variable scale.

sugerencia para solución.
```
    let dataShapes = svgchild.getAttribute("data-shapes");
    let scale = 1;
    let translateY = 0;
    isExportPNG ? scale = 2 : scale = 1;
    if (dataShapes === "c-block c-1 hat") {
      translateY = 20;
    }
    if (dataShapes === "hat") {
      translateY = 16;    
    }
    svgchild.setAttribute(
      "transform",
      `translate(0,${scale * translateY}) scale(${scale})`
    );
```
Borrar línea https://github.com/ScratchAddons/ScratchAddons/blob/9b145eb4e4623af9db72d9beb8a81bd10112c916/addons/blocks2image/userscript.js#L241 

Probar.

**Review by Samq64 — APPROVED** [review]

**Comment by WorldLanguages** [comment]
@Samq64 Right, we forgot about the cat ears

**Review by DNin01 — APPROVED** [review]
Tested on Edge. Looks good to me!

**Comment by DNin01** [comment]
> Acually the cat ears are cropped, but other than that everythng looks lood

That was already happening, though.

**Comment by Samq64** [comment]
In most cases if a PR is named Fix ... I would expect it to fix all occurrences of the bug even if a specific one wasn't documented. I'm not blocking merge any more, but I'm also not approving it.

**Comment by dos-pasos** [comment]
> @Samq64 Right, we forgot about the cat ears

que es el Cat Ears. No hablo inglés.

**Comment by WorldLanguages** [comment]
@dos-pasos Aquí hay imágenes: https://en.scratch-wiki.info/wiki/Cat_Block

**Comment by dos-pasos** [comment]
> @dos-pasos Aquí hay imágenes: https://en.scratch-wiki.info/wiki/Cat_Block

He hecho cambios para obtener resultados con los cat-blocks. Cree un fork del repo para blocks2image addons.
https://github.com/dos-pasos/ScratchAddons/blob/54e66d3385922c77909d67ccb7262c93262b2563/addons/blocks2image/userscript.js

Prueben si detectan errores.  Yo probé para mis necesidades y lo encuentro aceptable.

**Comment by WorldLanguages** [comment]
@dos-pasos Todos somos aficionados.

**Comment by WorldLanguages** [comment]
Recently, dos-pasos suggested more changes to support cat blocks. I'll check them soon.

**Comment by WorldLanguages** [comment]
> Recently, dos-pasos suggested more changes to support cat blocks. I'll check them soon.

For reference, I was referring to comment https://github.com/ScratchAddons/ScratchAddons/discussions/5601#discussioncomment-4677256

**Comment by BroJac5246** [comment]
Any progress?

This has been open for a long time. If cat blocks support isn't added soon then this should probably be merged and support should be added in a separate PR with a separate issue because this is a big bug in the addon.

**Comment by WorldLanguages** [comment]
> Any progress?

I'll try to find some time for this, but I encourage anyone reading this to push to my branch, PR to my fork or create a new PR on SA/SA. The code modifications are already described in the comment I linked to above, we just need to test them.

**Review by Samq64 — APPROVED** [review]

**Comment by Samq64** [comment]
@WorldLanguages @dos-pasos I fixed the cat ears.

File: addons/blocks2image/userscript.js [hunk 1]
```diff
@@ -57,6 +57,8 @@ export default async function ({ addon, console, msg }) {
     "data:image/svg+xml;base64,PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxMi43MSIgaGVpZ2h0PSI4Ljc5IiB2aWV3Qm94PSIwIDAgMTIuNzEgOC43OSI+PHRpdGxlPmRyb3Bkb3duLWFycm93PC90aXRsZT48ZyBvcGFjaXR5PSIwLjEiPjxwYXRoIGQ9Ik0xMi43MSwyLjQ0QTIuNDEsMi40MSwwLDAsMSwxMiw0LjE2TDguMDgsOC4wOGEyLjQ1LDIuNDUsMCwwLDEtMy40NSwwTDAuNzIsNC4xNkEyLjQyLDIuNDIsMCwwLDEsMCwyLjQ0LDIuNDgsMi40OCwwLDAsMSwuNzEuNzFDMSwwLjQ3LDEuNDMsMCw2LjM2LDBTMTEuNzUsMC40NiwxMiwuNzFBMi40NCwyLjQ0LDAsMCwxLDEyLjcxLDIuNDRaIiBmaWxsPSIjMjMxZjIwIi8+PC9nPjxwYXRoIGQ9Ik02LjM2LDcuNzlhMS40MywxLjQzLDAsMCwxLTEtLjQyTDEuNDIsMy40NWExLjQ0LDEuNDQsMCwwLDEsMC0yYzAuNTYtLjU2LDkuMzEtMC41Niw5Ljg3LDBhMS40NCwxLjQ0LDAsMCwxLDAsMkw3LjM3LDcuMzdBMS40MywxLjQzLDAsMCwxLDYuMzYsNy43OVoiIGZpbGw9IiNmZmYiLz48L3N2Zz4="
   );
 
+  const enabledAddons = await addon.self.getEnabledAddons("codeEditor");
```
Comment (WorldLanguages) [inline addons/blocks2image/userscript.js:60]: This only works properly because cat blocks can't be dynamically enabled/disabled, right?

Comment (Samq64) [reply addons/blocks2image/userscript.js:60]: Yeah I didn't consider dynamic enable but I can in case cat blocks supports it in the future.

Comment (WorldLanguages) [reply addons/blocks2image/userscript.js:60]: Let's hope we don't forget about this.

Comment (WorldLanguages) [reply addons/blocks2image/userscript.js:60]: I mean, this is still an improvement, so good job!

**Comment by WorldLanguages** [comment]
See PR #7644 for further changes to pixel-by-pixel rendering and offsets.

@DNin01 @Samq64 If possible, test that PR and check for any potential regressions. The test cases shown here should be of help.

File: addons/blocks2image/userscript.js [hunk 2]
```diff
@@ -175,10 +177,18 @@ export default async function ({ addon, console, msg }) {
...
-    );
+    let translateY = 0; // blocks no hat
+    const scale = isExportPNG ? 2 : 1;
+    if (dataShapes === "c-block c-1 hat") {
+      translateY = 20; // for My block
+    }
+    if (dataShapes === "hat") {
+      translateY = 16; // for Events
+      if (enabledAddons.includes("cat-blocks")) {
+        translateY += 16; // for cat ears
```
Comment (TheColaber) [inline addons/blocks2image/userscript.js:188]: Why doesn't this look at Blockly.BlockSvg.START_HAT_HEIGHT? Wouldn't using that remove the use for checking if cat blocks is enabled?

Comment (Samq64) [reply addons/blocks2image/userscript.js:188]: I don't know mucn about blockly. I can PR that if you want.

Comment (Samq64) [reply addons/blocks2image/userscript.js:188]: I there something similar for custom block hats?

Comment (WorldLanguages) [reply addons/blocks2image/userscript.js:188]: @TheColaber Send a pull request.

Comment (Samq64) [reply addons/blocks2image/userscript.js:188]: `START_HAT_HEIGHT` doesn't work for custom block hats. I've fixed single stack exporting locally with `getBBox()` but export all still needs some work.

Comment (WorldLanguages) [reply addons/blocks2image/userscript.js:188]: #7783

**Pull request merged**